src/components/menu/Menu.tsx

```diff
@@ -51,6 +51,8 @@ interface MenuProps {
     setRef?: Function;
     /** shouldOutlineFocus - whether the focused menu item should have an outline */
     shouldOutlineFocus?: boolean;
+    /** shouldStopPropagationOnClick - whether we should stop propagation on menu click */
+    shouldStopPropagationOnClick?: boolean;
```

Comment:
> The use case here is that we have an outer div which navigates to a different page on click. Therefore, when a menu item is clicked, we navigate to another page.
> 
> Do you think it makes more sense to just take in an `onClick` prop and allow the callers to define whether they want to use `stopPropagation` inside of their own `onClick` functions? Strangely, I notice that the `Menu` component here does have an `onClick` prop but its never called

That seems like it's more of an issue with the outer div than with these components. If more components (overlays, tooltips, banners, etc.) are added "above" this outer div, would all of those need this new prop as well?

is it possible to add a check in the outer div's onClick that verifies the click event is coming from the currentTarget instead of elements in which the event is propagating to it?

Reply:
That was our initial solution but there are many elements inside the div where do want the outer div click event to trigger. There's only 2 elements where we don't so it made more sense to just single those out instead